### PR TITLE
Feat: Add Terraform Azure Storage Account

### PR DESCRIPTION
**Pull request by maciejgwizdala**
### Description of your changes
Added support for Terraform Azure Storage Account

I have:

- [ ] Read and followed KubeVela's [contribution process](https://github.com/oam-dev/kubevela/blob/master/contribute/create-pull-request.md).
- [ ] [Related Docs](https://github.com/oam-dev/kubevela.io) updated properly. In a new feature or configuration option, an update to the documentation is necessary. 
- [ ] Run `make reviewable` to ensure this PR is ready for review.
- [ ] Added `backport release-x.y` labels to auto-backport this PR if necessary.

### CODE REVIEW

**Review by zzxwill — APPROVED** [review]
Great job!

**Comment by maciejgwizdala** [comment]
> Great job!

Thanks, that's exactly what I was missing in KubeVela. Great job with this project, guys! :)

In GH workflows there are just issues with
> toomanyrequests: You have reached your pull rate limit. You may increase the limit by authenticating and upgrading: https://www.docker.com/increase-rate-limit
I hope this will be solved soon, so PR could be merged.
This code is already tested with my own Azure Subscription and works fine.

**Comment by zzxwill** [comment]
@maciejgwizdala I will merge this great PR after one code owner approved it. 

BTW, would you like to add docs for this cloud resource?

You can create a similar PR like Azure Mariadb Server https://github.com/oam-dev/kubevela.io/pull/334/files.
The docs will show in https://kubevela.io/docs/next/end-user/components/cloud-services/terraform/azure-database-mariadb.

You can use `go run hack/references/generate.go` to generate the Markdown file.

**Comment by maciejgwizdala** [comment]
> @maciejgwizdala I will merge this great PR after one code owner approved it.
> 
> BTW, would you like to add docs for this cloud resource?
> 
> You can create a similar PR like Azure Mariadb Server https://github.com/oam-dev/kubevela.io/pull/334/files. The docs will show in https://kubevela.io/docs/next/end-user/components/cloud-services/terraform/azure-database-mariadb.
> 
> You can use `go run hack/references/generate.go` to generate the Markdown file.

PR for docs created: https://github.com/oam-dev/kubevela.io/pull/363
Sorry, I don't know Chinese, so I left English as placeholder.

File: vela-templates/addons/terraform/definitions/terraform-azure-storage-account.yaml
Comment (hongchaodeng) [inline vela-templates/addons/terraform/definitions/terraform-azure-storage-account.yaml:1]: @maciejgwizdala Could you add your name and contact as top level comment in this file? We would like to give you some credit. Meanwhile in case anyone has issues they can find some help

**Review by hongchaodeng — APPROVED** [review]
LGTM after nit

Could you also paste some pictures to show how it runs? It would be great to see the end result with some UI.

**Review by Somefive — APPROVED** [review]
lgtm

**Comment by wonderflow** [comment]
I'm merging this, feel free to send a new PR later .

**Pull request merged**
